File: Day 4 : Class vs. Instance/Solution.java
import java.util.*;

class Person
{
    int age;
    
    Person(int initialAge)
    {
        if (initialAge < 0)
        {
            this.age = 0;
            System.out.println("Age is not valid, setting age to 0.");
        }
        else
        {
            this.age = initialAge;   
        }
    }
    
    public void yearPasses()
    {
        this.age++;
    }
    
    public void amIOld()
    {
        if(this.age < 13)
        {
            System.out.println("You are young.");
        }
        else if(this.age >= 13 && this.age < 18)
        {
            System.out.println("You are a teenager.");
        }
        else
        {
            System.out.println("You are old.");
        }
    }
}

public class Solution
{
    public static void main(String[] args)
    {
        Scanner scan = new Scanner(System.in);
        
        int T = scan.nextInt();
        
        for(int i = 0; i < T; i++)
        {
            int age = scan.nextInt();
            Person p = new Person(age);
            p.amIOld();
            for(int j = 0; j < 3; j++)
            {
                p.yearPasses();
            }
            p.amIOld();
            System.out.println("");
        }
        
        scan.close();
    }
}

File: Day 4 : Class vs. Instance/Solution.ts
class Person {
    age: number
    constructor(initialAge: number) {
        if (initialAge >= 0) {
            this.age = initialAge;
        } else {
            console.log("Age is not valid, setting age to 0.");
            this.age = 0;
        }
    }

    yearPasses(): void {
        this.age++;
    }

    amIOld(): void {
        if (this.age < 13) {
            console.log("You are young.");
        } else if (this.age >= 13 && this.age < 18) {
            console.log("You are a teenager.");
        } else {
            console.log("You are old.");
        }
    }
}

function main() {
    const T: number = +readLine();
    for (let i = 0; i < T; i++) {
        const age: number = parseInt(readLine());
        const p: Person = new Person(age);
        p.amIOld();
        for (let j = 0; j < 3; j++) {
            p.yearPasses();
        }
        p.amIOld();
        console.log("");
    }
}